src/chatlog/chatlog.h

```diff
@@ -60,26 +60,26 @@ class ChatLog : public QGraphicsView
     bool hasTextToBeCopied() const;
 
     ChatLine::Ptr getTypingNotification() const;
-    QVector<ChatLine::Ptr> getLines();
+    QVector<ChatLine::Ptr> getLines() const;
     ChatLine::Ptr getLatestLine() const;
     // repetition interval sender name (sec)
     const uint repNameAfter = 5*60;
 
 signals:
-    void selectionChanged();
+    void selectionChanged() const;
 
 protected:
     QRectF calculateSceneRect() const;
     QRect getVisibleRect() const;
     ChatLineContent* getContentFromPos(QPointF scenePos) const;
 
-    void layout(int start, int end, qreal width);
+    void layout(int start, int end, qreal width) const;
     bool isOverSelection(QPointF scenePos) const;
     bool stickToBottom() const;
 
     qreal useableWidth() const;
 
-    void reposition(int start, int end, qreal deltaY);
+    void reposition(int start, int end, qreal deltaY) const;
```

Comment:
Semantically, I'm not sure if it makes sense to have "layout" and "reposition" be const.
If I have a const Chatlog& to someone, do I really want it to be possible to reposition and relayout things? Probably not.